### PR TITLE
Draft of deb-update-reprepro-repository

### PR DESCRIPTION
**Pull request by mih**
TODO:
- [x] Yield results for included package pieces
- [x] Implement support for .changes ingestion

Closes psychoinformatics-de/datalad-debian#54

### CODE REVIEW

File: datalad_debian/tests/test_update_reprepro_repository.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+@with_tempfile
+def test_update_reprepro_repo(path=None):
+    path = Path(path)
+    dist_ds_p = path / 'dist'
+    pkg_ds_p = path / 'dist' / 'packages' / 'tqdm'
+    archive_ds_p = path / 'archive'
+    # prepare a distribution, with a package
+    deb_new_distribution(dist_ds_p, **ckwa)
+    deb_new_package(dataset=dist_ds_p, name=pkg_ds_p.name, **ckwa)
+    # download a tiny package with little to no dependencies
```
Comment (aqw) [inline datalad_debian/tests/test_update_reprepro_repository.py:36]: Not sure if I agree with "no dependencies", as it depends on `python3-all` (and others) to build. ;-)

https://salsa.debian.org/python-team/packages/tqdm/-/blob/master/debian/control
```
Build-Depends: debhelper-compat (= 13),
               dh-python,
               python3-all,
               python3-docopt,
               python3-pandas <!nocheck>,
               python3-pytest <!nocheck>,
               python3-pytest-asyncio <!nocheck>,
               python3-pytest-timeout <!nocheck>,
               python3-rich <!nocheck>,
               python3-setuptools,
               python3-tk <!nocheck>,
```

Perhaps: "download a tiny python package with few dependencies"

File: setup.cfg [hunk 2]
```diff
@@ -16,6 +16,7 @@ python_requires = >= 3.7
 install_requires =
     datalad >= 0.17
     datalad-container
+    python-debian
```
Comment (aqw) [inline setup.cfg:19]: `python3-debian`?

File: datalad_debian/update_reprepro_repository.py [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+        # we can handle three types of files
+        # - changes files from builds of any kind
+        # - dsc of source packages
+        # - lonely debs
+        if r.get('state') in ('added', 'modified')
+        and Path(r['path']).suffix in ('.changes', '.dsc', '.deb')
+    ]
+    if not updated_files:
+        return
+    # TODO option to give a single commit across all updates?
```
Comment (aqw) [inline datalad_debian/update_reprepro_repository.py:157]: I think high commit count will be a problem for projects on the scale of Debian (given current limitations of git).

File: setup.cfg (hunk 2 above)
Comment (mih) [reply setup.cfg:19]: No, this is for pip, there is no `python3-debian`

File: datalad_debian/tests/test_update_reprepro_repository.py (hunk 1 above)
Comment (mih) [reply datalad_debian/tests/test_update_reprepro_repository.py]: I removed it. The number of dependencies is irrelevant for this implementation, nothing is built.

File: datalad_debian/update_reprepro_repository.py (hunk 3 above)
Comment (mih) [reply datalad_debian/update_reprepro_repository.py:158]: Yes. That is the rational behind the TODO

**Pull request merged**